workflow/rocoto/gfs_tasks.py

```diff
@@ -552,93 +552,163 @@ def _fcst_cycled(self):
-                          'ROTDIR': self._base.get('ROTDIR')}
-
+        postenvar_dict = {'FHRLST': 'f000',
+                          'UPP_RUN': 'analysis'}
         for key, value in postenvar_dict.items():
             postenvars.append(rocoto.create_envar(name=key, value=str(value)))
 
-        deps = []
         atm_anl_path = self._template_to_rocoto_cycstring(self._base["COM_ATMOS_ANALYSIS_TMPL"])
+        deps = []
+        data = f'{atm_anl_path}/{self.cdump}.t@Hz.atmanl.nc'
+        dep_dict = {'type': 'data', 'data': data, 'age': 120}
+        deps.append(rocoto.add_dependency(dep_dict))
+        data = f'{atm_anl_path}/{self.cdump}.t@Hz.sfcanl.nc'
+        dep_dict = {'type': 'data', 'data': data, 'age': 120}
+        deps.append(rocoto.add_dependency(dep_dict))
         data = f'{atm_anl_path}/{self.cdump}.t@Hz.loganl.txt'
-        dep_dict = {'type': 'data', 'data': data}
+        dep_dict = {'type': 'data', 'data': data, 'age': 60}
+        deps.append(rocoto.add_dependency(dep_dict))
+        dependencies = rocoto.create_dependency(dep=deps, dep_condition='and')
+        resources = self.get_resource('upp')
+        task = create_wf_task('atmanlupp', resources, cdump=self.cdump, envar=postenvars, dependency=dependencies,
+                              cycledef=self.cdump, command='&JOBS_DIR;/upp.sh')
+
+        return task
+
+    def atmanlprod(self):
+        postenvars = self.envars.copy()
+        postenvar_dict = {'FHRLST': '-f001'}
+        for key, value in postenvar_dict.items():
+            postenvars.append(rocoto.create_envar(name=key, value=str(value)))
+
+        atm_master_path = self._template_to_rocoto_cycstring(self._base["COM_ATMOS_MASTER_TMPL"])
+        deps = []
+        data = f'{atm_master_path}/{self.cdump}.t@Hz.master.grb2anl'
+        dep_dict = {'type': 'data', 'data': data, 'age': 120}
         deps.append(rocoto.add_dependency(dep_dict))
         dependencies = rocoto.create_dependency(dep=deps)
-        resources = self.get_resource('post')
-        task = create_wf_task('postanl', resources, cdump=self.cdump, envar=postenvars, dependency=dependencies,
-                              cycledef=self.cdump)
+        resources = self.get_resource('atmos_products')
+        task = create_wf_task('atmanlprod', resources, cdump=self.cdump, envar=postenvars, dependency=dependencies,
+                              cycledef=self.cdump, command='&JOBS_DIR;/atmos_products.sh')
 
         return task
 
-    def ocnpost(self):
-        if self.app_config.mode in ['forecast-only']:  # TODO: fix ocnpost in cycled mode
-            return self._post_task('ocnpost')
+    @staticmethod
+    def _get_ufs_postproc_grps(cdump, config):
 
-    def _post_task(self, task_name):
-        if task_name not in ['post', 'ocnpost']:
-            raise KeyError(f'Invalid post-processing task: {task_name}')
+        fhmin = config['FHMIN']
+        fhmax = config['FHMAX']
+        fhout = config['FHOUT']
```

Comment:
```suggestion
        suffix = '_GFS' if cdump in ['gfs'] else ''
        fhmin = config[f'FHMIN{suffix}']
        fhout = config[f'FHOUT{suffix}']
```

Reply:
I don't think this makes the logic any clearer, since with `_GFS`, there is the added complication of `_HF`.
For the purposes of clarity, I suggest we keep this unchanged (for now).